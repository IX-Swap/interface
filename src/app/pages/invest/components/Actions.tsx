import React from 'react'
import { Commitment } from 'types/commitment'
import { AppRouterLinkComponent } from 'components/AppRouterLink'
import { Button } from '@material-ui/core'
import { CommitmentRoute } from 'app/pages/invest/router/config'

export interface ActionsProps {
  item: Commitment
}

export const Actions = ({ item }: ActionsProps) => {
  return (
    <Button
      component={AppRouterLinkComponent}
      to={CommitmentRoute.view}
      params={{ commitmentId: item._id }}
      size='small'
      variant='outlined'
    >
      View
    </Button>
  )
}
