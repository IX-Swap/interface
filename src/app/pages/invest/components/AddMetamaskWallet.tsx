import { Link, Typography } from '@mui/material'
import { AccountsRoute } from 'app/pages/accounts/router/config'
import React from 'react'

export const AddMetamaskWallet = () => {
  return (
    <>
      <Link
        style={{ backgroundColor: 'transparent', textTransform: 'none' }}
        variant='body1'
        type='button'
        target='_blank'
        href={AccountsRoute.withdrawalAddressesCreate}
      >
        <Typography color='primary'>Add Your Metamask Wallet</Typography>
      </Link>
    </>
  )
}
