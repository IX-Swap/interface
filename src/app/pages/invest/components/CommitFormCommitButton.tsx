import React from 'react'
import { useCommitmentValidator } from 'app/pages/invest/hooks/useCommitmentValidator'
import { useFormContext } from 'react-hook-form'
import { Button } from '@material-ui/core'
import { CommitmentFormValues } from 'types/commitment'
import { useMakeCommitment } from 'app/pages/invest/hooks/useMakeCommitment'

export interface CommitmentFormCommitButtonProps {
  assetId: string
  minInvestment: number | null
  dsoId: string
  currency: string
}

export const CommitmentFormCommitButton = ({
  assetId,
  minInvestment,
  dsoId,
  currency
}: CommitmentFormCommitButtonProps) => {
  const { isValid } = useCommitmentValidator({ assetId, minInvestment })
  const { handleSubmit } = useFormContext()
  const {
    commit: [makeCommitment]
  } = useMakeCommitment()

  const submit = handleSubmit(async (data: CommitmentFormValues) => {
    await makeCommitment({
      ...data,
      withdrawalAddress:
        data.withdrawalAddress === '' ? undefined : data.withdrawalAddress,
      signedSubscriptionDocument: data.signedSubscriptionDocument?._id,
      dso: dsoId,
      currency
    })
  })

  const handleClick = () => {
    void submit()
  }

  return (
    <Button
      onClick={handleClick}
      data-type='commit'
      color='primary'
      variant='outlined'
      fullWidth
      disabled={!isValid}
      type='button'
    >
      Commit
    </Button>
  )
}
