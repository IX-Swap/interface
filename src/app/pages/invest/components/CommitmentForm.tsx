import React, { PropsWithChildren } from 'react'
import { CommitmentFormValues } from 'types/commitment'
import { FormProps, Form } from 'components/form/Form'
import { useMakeCommitment } from 'app/pages/invest/hooks/useMakeCommitment'
import { commitmentFormValidationSchema } from 'app/pages/invest/validation'

export interface CommitmentFormProps {
  dso: string
  currency: string
}

export const CommitmentForm = (
  props: PropsWithChildren<
    CommitmentFormProps & FormProps<CommitmentFormValues>
  >
) => {
  const { dso, currency, children, ...rest } = props
  const [makeInvestment] = useMakeCommitment()
  const handleSubmit = async ({
    totalAmount,
    pricePerUnit,
    withdrawalAddress,
    ...values
  }: CommitmentFormValues) => {
    await makeInvestment({
      ...values,
      withdrawalAddress:
        withdrawalAddress === '' ? undefined : withdrawalAddress,
      signedSubscriptionDocument: values.signedSubscriptionDocument._id,
      dso,
      currency
    })
  }

  return (
    <Form
      {...rest}
      onSubmit={handleSubmit}
      validationSchema={commitmentFormValidationSchema}
    >
      {children}
    </Form>
  )
}
