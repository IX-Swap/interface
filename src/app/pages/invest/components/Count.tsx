import React from 'react'
import { Typography } from '@mui/material'

export interface CountProps {
  value: string | number
}

export const Count = ({ value }: CountProps) => {
  return (
    <Typography
      display={'inline'}
      sx={theme => ({
        backgroundColor: '#4c88ff10',
        color: theme.palette.info.main,
        borderRadius: 16,
        marginLeft: 1.5,
        px: 1,
        py: 0.5
      })}
    >
      {value ?? 0}
    </Typography>
  )
}
