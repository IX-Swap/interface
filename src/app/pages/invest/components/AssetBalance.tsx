import React from 'react'
import { useBalancesByAssetId } from 'hooks/balance/useBalancesByAssetId'
import { LabelledValue } from 'components/LabelledValue'
import { formatMoney } from 'helpers/numbers'

export interface AssetBalanceProps {
  assetId: string
  symbol: string
}

export const AssetBalance = (props: AssetBalanceProps) => {
  const { assetId, symbol } = props
  const { data, isLoading, isIdle } = useBalancesByAssetId(assetId)

  if (isLoading || isIdle) {
    return null
  }

  return (
    <LabelledValue
      label='Account Balance'
      value={formatMoney(
        data.map[assetId]?.available ?? 0,
        data.map[assetId]?.symbol ?? symbol
      )}
    />
  )
}
