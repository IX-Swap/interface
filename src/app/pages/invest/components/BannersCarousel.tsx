import React from 'react'
import { Box, Grid, Typography } from '@mui/material'
import { useTheme } from '@mui/material/styles'
import { useBannersList } from 'app/pages/admin/hooks/useBannersList'
import { Avatar } from 'components/Avatar'
import { Carousel } from 'react-responsive-carousel'
import 'react-responsive-carousel/lib/styles/carousel.min.css'
import { DataroomBanner } from 'types/dataroomBanner'

export const BannersCarousel = () => {
  const theme = useTheme()
  const { data, isLoading } = useBannersList()

  if (typeof data === 'undefined' || isLoading) {
    return null
  }

  return (
    <Grid
      style={{ paddingTop: theme.spacing(3), paddingBottom: theme.spacing(4) }}
    >
      <Carousel
        showThumbs={false}
        showArrows={false}
        showStatus={false}
        showIndicators={false}
        interval={3000}
        infiniteLoop
        autoPlay
      >
        {data.map((item: DataroomBanner) => {
          const { title, _id } = item

          return (
            <div key={_id}>
              {title !== '' ? (
                <Typography
                  variant='h4'
                  style={{
                    marginBottom: theme.spacing(2.5),
                    textAlign: 'left'
                  }}
                >
                  {title}
                </Typography>
              ) : null}
              <Avatar
                size={['100%', 'auto']}
                documentId={_id}
                variant={'square'}
                type={'banner'}
                fallback={<Box />}
              />
            </div>
          )
        })}
      </Carousel>
    </Grid>
  )
}
