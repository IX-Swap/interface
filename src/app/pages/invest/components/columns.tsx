import React from 'react'
import { TableColumn } from 'types/util'
import { formatMoney, formatAmount } from 'helpers/numbers'
import { Commitment } from 'types/commitment'
import { renderStatusColumn } from 'app/pages/authorizer/hooks/useAuthorizerView'
import { DSOLogo } from 'app/components/DSO/components/DSOLogo'

export const renderCommitmentMoney = (a: number, row: Commitment) =>
  formatMoney(a, row.currency.numberFormat.currency)

export const renderCommitmentAvatar = (a: string, row: Commitment) => (
  <DSOLogo dsoId={row.dso._id} size={40} />
)

export const columns: Array<TableColumn<Commitment>> = [
  {
    key: '_id',
    label: '',
    render: renderCommitmentAvatar
  },
  {
    key: 'dso.tokenSymbol',
    label: 'Name'
  },
  {
    key: 'pricePerUnit',
    label: 'Unit Price',
    align: 'right',
    headAlign: 'right',
    render: renderCommitmentMoney
  },
  {
    key: 'dso.capitalStructure',
    label: 'Capital Structure'
  },
  {
    key: 'totalAmount',
    label: 'Investment Amount',
    align: 'right',
    headAlign: 'right',
    render: renderCommitmentMoney
  },
  {
    key: 'numberOfUnits',
    label: 'Number of Digital Securities',
    align: 'right',
    headAlign: 'right',
    render: formatAmount
  },
  {
    key: 'status',
    label: 'Status',
    render: renderStatusColumn
  }
]

export default columns
