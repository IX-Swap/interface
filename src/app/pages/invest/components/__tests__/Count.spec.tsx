import React from 'react'
import { render } from 'test-utils'
import { Count } from 'app/pages/invest/components/Count'

describe('Count', () => {
  afterEach(async () => {
    jest.clearAllMocks()
  })

  it('renders correct value if value is undefined', () => {
    const { container } = render(<Count value={undefined as any} />)
    expect(container).toHaveTextContent('0')
  })

  it('renders correct number value', () => {
    const value = 32
    const { container } = render(<Count value={value} />)

    expect(container).toHaveTextContent(value.toString())
  })

  it('renders correct string value, should match snapshot', () => {
    const value = '32'
    const { container } = render(<Count value={value} />)

    expect(container).toHaveTextContent(value)
    expect(container).toMatchSnapshot()
  })
})
