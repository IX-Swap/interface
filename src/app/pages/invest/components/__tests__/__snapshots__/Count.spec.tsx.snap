// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Count renders correct string value, should match snapshot 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1vpnri2-MuiTypography-root"
  >
    32
  </p>
</div>
`;
